vllm/platforms/cpu.py

```diff
@@ -148,6 +149,13 @@ def check_and_update_config(cls, vllm_config: VllmConfig) -> None:
         # To hint IPEX uses shared memory based AllReduce
         os.environ["LOCAL_WORLD_SIZE"] = str(
             vllm_config.parallel_config.tensor_parallel_size)
+        if sys.platform == "darwin" and \
+                envs.VLLM_WORKER_MULTIPROC_METHOD == "fork":
+            if os.environ.get('VLLM_WORKER_MULTIPROC_METHOD', None) is None:
+                logger.warning(
+                    "Default to spawn method on MacOS. If this is not desired,"
+                    " set VLLM_WORKER_MULTIPROC_METHOD to fork explicitly.")
+                os.environ['VLLM_WORKER_MULTIPROC_METHOD'] = 'spawn'
```

Comment:
This would affect more than tensor parallel. It would also affect the default behavior of `vllm serve`, where we run the API server in a different process than the engine core. Is it working without tensor parallel?

Reply:
because multiprocessing issues come up regularly and the conflicts can be a lot to keep track of, we have a doc on this topic. It would be good to get to the bottom of this behavior on mac and update this doc with the findings: https://docs.vllm.ai/en/latest/design/multiprocessing.html

Reply:
I have tested it and after setting it up, it works perfectly under macOS without TP.
If you have concerns, you don't have to set this value (of course, the environment variable has a higher priority). This is just to make VLLM work out of the box on macOS without having to consult the docs every time.

Reply:
Of course, I can remove this code by stating so in the documentation.

Reply:
it seems to me, macos uses spawn by default: https://docs.python.org/3/library/multiprocessing.html

Reply:
`python3 -c 'import multiprocessing; print(multiprocessing.get_start_method())'` gives `spawn`.

so we are forcing `fork` by default even in macos, and this pr fixes it.

of course, we can document it in [docs.vllm.ai/en/latest/design/multiprocessing.html](https://docs.vllm.ai/en/latest/design/multiprocessing.html)

Reply:
I added it, you can take a look. Also, I saw that [docs/readthedocs.org:vllm](https://readthedocs.org/projects/vllm/builds/27663455/) failed, but it seems to be unrelated to the PR content. Should I try again to solve it? But I can't do it.